Updating Darwinbuild and Darwinxref

csekel opened the pull request:
By submitting a request, you represent that you have the right to license
your contribution to the community, and agree that your contributions are
licensed under the [BSD License accompanying darwinbuild](LICENSE.txt).

For existing files modified by your request, you represent that you have
retained any existing copyright notices and licensing terms. For each new
file in your request, you represent that you have added to the file a
copyright notice (including the year and the copyright owner's name) and
darwinbuild's licensing terms.

Updating Darwinbuild and Darwinxref
These updates fix issues within Darwinbuild program that would allow
MacPorts and the original project to compile and run on 10.10/10.11
currently 10.12 is unsupported with this updated commit.

Echo was missing before darwinxref invocation;
Allow http://opensource.apple.com/tarballs/ as sources_site as
src.macosforge.org/Projects is not redirected anymore;
Don't change filename when fetching tarballs from apple;
Replacing deprecated CFPropertyListCreateFromXMLData when osx > SL ;
Replacing deprecated CFPropertyListCreateXMLData when osx > SL

csekel committed:
Updating Darwinbuild and Darwinxref

Updating Darwinbuild and Darwinxref
These updates fix issues within Darwinbuild program that would allow
MacPorts and the original project to compile and run on 10.10/10.11
currently 10.12 is unsupported with this updated commit.

Echo was missing before darwinxref invocation;
Allow http://opensource.apple.com/tarballs/ as sources_site as
src.macosforge.org/Projects is not redirected anymore;
Don't change filename when fetching tarballs from apple;
Replacing deprecated CFPropertyListCreateFromXMLData when osx > SL ;
Replacing deprecated CFPropertyListCreateXMLData when osx > SL

csekel committed:
Modified to Build on OS X 10.12

Modified to Build on OS X 10.12

csekel committed:
Fixed Errors in Darwinbuild

csekel committed:
fixed dirty patch.

csekel committed:
Fixed patch

csekel committed:
fixed patch

bhuntsman commented:
Do you think you could use a .gitignore to avoid picking up all the per-user crap that Xcode creates?  Something like this:

```
.gitignore
build/
.DS_Store
darwinbuild.xcodeproj/project.xcworkspace/
darwinbuild.xcodeproj/xcuserdata/
```

Only the project.pbxproj file matters...

Thank you for your contributions!

csekel committed:
Fixed line 215

csekel commented:
@bhuntsman yes I will do that

csekel committed:
Added .gitignore

csekel committed:
Added Feature to Darwinbuild

Added patch_sites like how you can added source_sites but for patches
of source codes.

Example!!
<key>patch_sites</key>
  <array>
    <string>http://6X.2XX.XX5.1XX/Patches</string>
  </array>
